Sprint 2

Timur233 opened the pull request.

Timur233 committed:
Перевел проект на TS

Timur233 committed:
Добавил в проект файл .editorconfig

Timur233 committed:
Добавил eslint в проект

Timur233 committed:
Добавил в проект stylelint

Timur233 committed:
Доработки:
 - Рендеринг
 - Обработка событий элементов
 - Работа с просами
 - Работа с дочерними компонентами

Timur233 committed:
Мучительно внедрил компонентный подход

Timur233 committed:
Правки по компонентам

Timur233 committed:
Удалил лишние стили

Timur233 committed:
Правки
- Добавил HTTPTransport класс
- Добавил страницу для работы с классом
- Исправил все косяки по структуре файлов и наименованиям
- Прошел по коду, исправил ошибки по типам и
в обще привел весь код в порядок

Timur233 committed:
README

Timur233 committed:
Ошибка развертывания netlify

Timur233 committed:
Добавил роут на страницу http-playground

Timur233 committed:
Удалил vite из зависимостей

Timur233 committed:
Добавил typescript в зависимости

Timur233 committed:
Обновил eslint parser и eslint pugin

Timur233 committed:
Обновил eslint parser и eslint pugin

Timur233 committed:
Удалил vite из зависимостей

Timur233 committed:
Обновил eslint parser и eslint pugin

Timur233 committed:
Исправил правила eslint

Timur233 commented:
Ой накрутил я с настройками линтеров)) 
Надеюсь не составит проблем. Спасибо!

mikebars1995 commented on src/services/Component.ts:
```diff
@@ -0,0 +1,225 @@
+} from './types.ts';
+import PropsManager from './PropsManager.ts';
+
+export default class Component {
```
**Можно лучше**

лучше всего передавать тип пропсов в виде дженерика в `Component `, чтобы можно было конкретный тип пропсов всегда иметь под каждый отдельный экземпляр класса `Component `. Будет очень универсально 

```js
abstract class Component <Props extends Record<string, any> = unknown> {

```

mikebars1995 commented on src/services/Component.ts:
```diff
@@ -0,0 +1,225 @@
+        this._eventBus().emit(Component.EVENTS.FLOW_CDU, prevProps, nextProps);
+    }
+
+    private removeEvents():void {
```
**Отлично**


Отлично, что не забываете про удаление обработчиков

mikebars1995 commented on src/services/HTTPtransport.ts:
```diff
@@ -0,0 +1,94 @@
+class HTTPTransport {
+    METHODS = METHODS;
+
+    get(url: string, options?: Options) {
```
**Можно лучше**
Если типизация повторяется в каждом методе, то нужно типизировать сам метод, а не дублировать типизацию аргументов. Код станет чище

```jsx

// создаем тип метода
type HTTPMethod = (url: string, options?: Options) => Promise<unknown>

export class HTTPTransport {

  // используем тип и удаляем дублирование в аргументах
  get: HTTPMethod = (url, options = {}) => (
    this.request(url, {...options, method: METHODS.GET}, options.timeout)
  )
  // используем тип и удаляем дублирование в аргументах
  put: HTTPMethod = (url, options = {}) => (
    this.request(url, {...options, method: METHODS.PUT}, options.timeout)
  )
  // используем тип и удаляем дублирование в аргументах
  post: HTTPMethod = (url, options = {}) => (
    this.request(url, {...options, method: METHODS.POST}, options.timeout)
  )
  // используем тип и удаляем дублирование в аргументах
  delete: HTTPMethod = (url, options = {}) => (
    this.request(url, {...options, method: METHODS.DELETE}, options.timeout)
  )

```

mikebars1995 reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Посмотрите на *гитхабе* все комментарии к коду (нужно прокрутить вниз страницу там)

Работа проделана огромная


Что сделано отлично:


* `Readme` хорошо оформлен
* Проект задеплоен и отлично работает
* Отлично, что не забыли про `.gitignore`
* Отлично, что ограничиваете версию `node`
* Отлично, что нет `EOF` ошибок в гите
* Хорошая структура папок и файлов
* Добавлен `ESLint` и `Stylelint`
* Отлично, что цвета вынесены в `css`-константы
* Отлично, что не забываете про `alt` в тегах `img`
* Отлично, что не забываете про тег `main` для главной части страницы
* Отлично, что не забываете про удаление обработчиков
* Отлично, что показываете информативные сообщения валидации



Что можно улучшить:

* лучше всего передавать тип пропсов в виде дженерика в `Component `, чтобы можно было конкретный тип пропсов всегда иметь под каждый отдельный экземпляр класса `Component `. Будет очень универсально 
* Если типизация повторяется в каждом методе, то нужно типизировать сам метод, а не дублировать типизацию аргументов. Код станет чище



Поздравляю! Ваша работа принята.

Критических замечаний нет

Вы отлично потрудились.

Удачного дальнейшего обучения.

Timur233 committed:
Доработки по ревью
- Поправил ошибки в реадми
- Сделал типизацию методов class HTTPtransport.ts

Timur233 committed:
Правки:
- Исправил анимацию searchbar'a, не работала в
сафари переделал показ и скрытие placeholder'а на js

Timur233 committed:
Правки:
- Исправил отображение иконок в закреплении файла
(не отображались в сафари)

Timur233 committed:
Правки

Timur233 committed:
Вынес тип данных СomponentData в generic
